02-intermediate-ruby/class-methods-and-self.md

```diff
@@ -10,171 +10,209 @@
+But what if we have things that related to **all** `Song`s? Or setting up a bunch of `Song`s? Stay tuned...!
 
-## Class Methods & `self`
+## Class Methods and `self`
 
 Class methods are methods that are called on a **class** and instance methods are methods that are called on an **instance of a class**.
 
-### Let's Try It
+### Class Method Syntax
 
 With no specific problem in mind to start, let's see what the difference is in the **class definition**:
+
 ```ruby
-class TryingThisOut
-  def self.cool
+class ClassName
+  def self.class_method
     "This is a CLASS METHOD"
   end
 
-  def rad
+  def instance_method
     "This is an INSTANCE METHOD"
   end
 end
 ```
 
 **Class methods** are _defined_ like **instance methods**, but they start with `self.` and are called internally with the `self` identifier.
 
-Class methods are called directly by the _class_ and not by an _instance_ of the class.
+Class methods are called directly on the _class_ and not on an _instance_ of the class.
+
 ```ruby
-puts TryingThisOut.cool
-# will output 'This is a CLASS METHOD'
+puts ClassName.class_method
+# 'This is a CLASS METHOD'
 
-test = TryingThisOut.new
-puts test.rad
-# will output 'This is an INSTANCE METHOD'
+instance = ClassName.new
+puts instance.instance_method
+# 'This is an INSTANCE METHOD'
 ```
 
-### Adding in Context
+We've already seen one class method: `new`. `new` is a built-in Ruby method defined on all objects, which builds an instance of that class, calls its `initialize` method, and returns it.
+
+When writing about a class method, we use a dot instead of a pound sign: `ClassName.class_method`. Note that this matches the way the method is called.
+
+### Adding Class Methods to `Song`
 
 Now that we see the syntax for how we'd use a **class method** versus an **instance method** let's see why we'd want to use one over the other.
 
+Let's think back to the `Song` class we created earlier. We'll start with tracking the total number of plays across all songs. For this we'll need to add a _class variable_, `@@total_plays`, as well as a method to read its value, `Song.total_plays`.
 
-#### Revisit our `Penguin`
-Let's think back to the `Penguin` class we created from homework previously.
 ```ruby
-class Penguin
-  attr_reader :name
+class Song
+  attr_reader :title, :artist, :filename, :play_count
+
+  # Initialize our total play count
+  # This will be set to 0 when the program is loaded
+  @@total_plays = 0
+
+  def initialize(title, artist, filename)
+    @title = title
+    @artist = artist
+    @filename = filename
+    @play_count = 0
+  end
 
-  def initialize(name)
-    @name = name
+  def summary
+    return "#{@title}, by #{@artist}"
   end
 
-  def looks
-    return "Fuzzy!"
+  def play
+    @play_count += 1
+    @@total_plays += 1
+    # ... load the song data from the file and send it to the speakers ...
+  end
+
+  # Define a class method to access the class variable
+  def self.total_plays
+    return @@total_plays
```

Comment:
Right now it seems like this is just accessing a variable, so it might be good to wrap it in a string similar to the summary method so it is doing more than just returning a variable?